src/analyses/call_graph.h

```diff
@@ -25,6 +25,17 @@ class call_grapht
   explicit call_grapht(const goto_modelt &, bool collect_callsites=false);
   explicit call_grapht(const goto_functionst &, bool collect_callsites=false);
 
+  // These two constructors build a call graph restricted to functions
+  // reachable from the given root.
+  call_grapht(
```

Comment:
We now have a lot of constructors - it might be worth considering a collection of named static functions instead? `create_callgraph_from_root_node`? Maybe even a builder pattern so the common parts can be setup first (e.g. `collect_callsites`).